### PR TITLE
feat(core): Sync Streams with cache before returning from multiQuery

### PR DESCRIPTION
**Pull request by stbrody**
This ensures returned Streams have considered tips learned about from CommitIDs in the same batch

This PR was inspired by this comment: https://github.com/ceramicnetwork/ceramic-anchor-service/pull/473#discussion_r667321534

### CODE REVIEW

File: packages/core/src/ceramic.ts [hunk 1]
```diff
@@ -662,15 +662,27 @@ class Ceramic implements CeramicApi {
...
+    const results = queryResults.reduce((acc, res) => ({ ...acc, ...res }), {})
+    // Before returning the results, sync each Stream to its current state in the cache.  This is
+    // necessary to handle the case where there are two MultiQueries in this batch for the same stream,
+    // one on a specific CommitID and one on a base StreamID, and loading the CommitID tells this
+    // node about a tip it had not previously heard about that turns out to be the best current tip.
+    // This ensures the returned Streams always are as up-to-date as possible, and is behavior that
+    // the anchor service relies upon.
+    await Promise.all(Object.values(results).map((stream) => {
+      // We swallow errors because if any of the queries used CommitIDs or the `atTime` arg calling
+      // sync() on them will throw.
```
Comment (oed) [inline packages/core/src/ceramic.ts:682]: Don't we have a way to check if a stream is read only? I thought so.

Comment (oed) [reply packages/core/src/ceramic.ts:682]: If not it would make sense to add a `isReadOnly` property!

**Review by oed — COMMENTED** [review]
Looks good, one suggestion.

File: packages/core/src/ceramic.ts (hunk 1 above)
Comment (stbrody) [reply packages/core/src/ceramic.ts]: yep, that's definitely better, good call

File: packages/common/src/stream.ts [hunk 2]
```diff
@@ -228,7 +228,12 @@ export abstract class Stream extends Observable<StreamState> implements StreamSt
    * Makes this stream read-only. After this has been called any future attempts to call
    * mutation methods on the instance will throw.
    */
-  abstract makeReadOnly()
+  abstract makeReadOnly(): void
+
+  /**
+   * Returns true if 'makeReadOnly()' has been called previously.
+   */
+  abstract isReadOnly(): boolean
```
Comment (oed) [inline packages/common/src/stream.ts:236]: Why is this a method vs a property with a getter?

**Review by oed — APPROVED** [review]
lgtm. Added a question about the api, but I'm fine either way.

File: packages/common/src/stream.ts (hunk 2 above)
Comment (stbrody) [reply packages/common/src/stream.ts:236]: honestly no reason. Can change

**Pull request merged**